Tools/Scripts/webkitpy/common/system/executive.py

```diff
@@ -389,37 +389,42 @@ def run_command(self,
                              env=env,
                              close_fds=True,
                              pass_fds=pass_fds)
-        with process:
+        try:
+            output = ''
             if not string_to_communicate:
-                output = process.communicate()[0]
+                output = process.communicate(timeout=Timeout.difference())[0]
             else:
-                output = process.communicate(string_utils.encode(string_to_communicate, encoding='utf-8'))[0]
+                output = process.communicate(string_utils.encode(string_to_communicate, encoding='utf-8'), timeout=Timeout.difference())[0]
 
             # run_command automatically decodes to unicode() and converts CRLF to LF unless explicitly told not to.
             if decode_output:
                 output = string_utils.decode(output, encoding=self._child_process_encoding()).replace('\r\n', '\n')
 
             # wait() is not threadsafe and can throw OSError due to:
             # http://bugs.python.org/issue1731717
-            exit_code = process.wait()
+            exit_code = process.wait(timeout=Timeout.difference())
 
             _log.debug('"%s" took %.2fs' % (self.command_for_printing(args), time.time() - start_time))
 
-            if return_exit_code:
-                return exit_code
+        except subprocess.TimeoutExpired:
+            _log.debug('"%s" timed out after %.2fs' % (self.command_for_printing(args), time.time() - start_time))
+            exit_code = 255
 
-            if exit_code:
-                script_error = ScriptError(script_args=args,
-                                           exit_code=exit_code,
-                                           output=output,
-                                           cwd=cwd)
+        finally:
+            if process.poll() is None:
+                process.kill()
```

Comment:
Do we want to wait for it to actually be killed? (IIRC, on NT systems, killing a process can take a small amount of time.)